spikeinterface/sortingcomponents/clustering/clustering_tools.py

```diff
@@ -593,7 +589,9 @@ def remove_duplicates_via_matching(waveform_extractor, noise_levels, peak_labels
     labels = np.unique(new_labels)
     labels = labels[labels>=0]
 
-    shutil.rmtree(tmp_folder)
+    del recording, sub_recording
+    import os
+    os.remove(tmp_filename)
```

Comment:
```suggestion
    os.remove(tmp_filename)
    os.rmdir(tmp_folder)
```
Can you move the import at the top? 

Also, this will leave the empty `tmp_folder` there..maybe you can add an `os.rmdir()`?